### ISSUE TITLE
Connect-AZAccount : The provided account azureuser@contoso.com does not have access to subscription ID "xxxx-xxxx-xxxx-xxxx". Please try logging in with different credentials or a different subscription ID

### ISSUE THREAD

**Issue by andy68man** [MicrosoftDocs/partner-center-docs-powershell#122]
Hello,

We have a CSP model and are trying to scrape all the virtual machines from our customer tenants using the consent from the web app hosted in our CSP.

I have successfully been able to create a refresh token and connect to the partner portal using Connect-MsolService to get a list of tenants and subscriptions, but I cannot connect to the customers Azure subscriptions using the example given. When I try and connect to the customer subscription I get the following error:
Connect-AZAccount : The provided account azureuser@contoso.com does not have access to subscription ID "xxxx-xxxx-xxxx-xxxx". Please try logging in with different credentials or a different subscription ID

If I try and connect manually using the same account and use MFA it works fine.

The PowerShell script is as follows:
```
$password = "Web App secret key" | ConvertTo-SecureString -asPlainText -Force
$username = "App ID"
$cred = new-object -typename System.Management.Automation.PSCredential -argumentlist $username, $password
$UserCredential = Get-Credential $cred
$refreshToken = ''
$graphToken = New-PartnerAccessToken -ApplicationId $username -Credential $UserCredential -RefreshToken $refreshToken -Scopes 'https://graph.microsoft.com/.default' -ServicePrincipal -Tenant 'yyyy-yyyy-yyyy-yyyy'
$azureToken = New-PartnerAccessToken -ApplicationId $username -Credential $UserCredential -RefreshToken $refreshToken -Scopes 'https://management.azure.com/user_impersonation' -ServicePrincipal -Tenant 'yyyy-yyyy-yyyy-yyyy'

Connect-AZAccount -verbose -AccountId 'azureuser@contoso.com' -AccessToken $azureToken.AccessToken -GraphAccessToken $graphToken.AccessToken -Tenant 'xxxx-xxxx-xxxx-xxxx' -Subscription 'xxxx-xxxx-xxxx-xxxx'
```

Regards,

Andy

**Comment by andy68man** [comment]
Hello,

Anyone got any idea why my CSP Admin account doesn't have permission to view customer tenant subscriptions? I can get to it fine by connecting manually, are there any logs or anything I can look at?

Thanks,

Andy

**Comment by isaiahwilliams** [comment]
@andy68man does the user account you are using have Admin Privileges? If it does can you try connecting using the following 

```powershell
$password = "Web App secret key" | ConvertTo-SecureString -asPlainText -Force
$username = "App ID"
$cred = new-object -typename System.Management.Automation.PSCredential -argumentlist $username, $password
$UserCredential = Get-Credential $cred
$refreshToken = ''
$graphToken = New-PartnerAccessToken -ApplicationId $username -Credential $UserCredential -RefreshToken $refreshToken -Scopes 'https://graph.windows.net/.default' -ServicePrincipal -Tenant 'yyyy-yyyy-yyyy-yyyy'
$azureToken = New-PartnerAccessToken -ApplicationId $username -Credential $UserCredential -RefreshToken $refreshToken -Scopes 'https://management.azure.com/user_impersonation' -ServicePrincipal -Tenant 'yyyy-yyyy-yyyy-yyyy'

Connect-AZAccount -verbose -AccountId 'azureuser@contoso.com' -AccessToken $azureToken.AccessToken -GraphAccessToken $graphToken.AccessToken -Tenant 'xxxx-xxxx-xxxx-xxxx' -Subscription 'xxxx-xxxx-xxxx-xxxx'
```

**Comment by andy68man** [comment]
Hi Isaiah,

Thanks for answering my ticket, I found the issue this morning - I was using the built in Web App in the partner portal which by default only has access to the top CSP organisation instead of multi organisation.

I followed this guide: 
https://www.microsoftpartnercommunity.com/t5/Secure-Application-Model/Exchange-Online-and-the-Secure-App-Model/td-p/11771

This lead to a PowerShell that created a new Web App with the correct settings, and then I had to give it the Azure Service Management permissions and I was then able to connect fine.

Thanks for taking a look at my issue!

Regards,

Andy
